plugins/Referrers/Columns/Base.php

```diff
@@ -137,6 +139,45 @@ protected function getReferrerInformation($referrerUrl, $currentUrl, $idSite, Re
         return $referrerInformation;
     }
 
+    protected function excludeQueryParamsFromReferrerUrl()
+    {
+        // Add additional excludes for specific referrers
+        Piwik::addAction('Tracker.PageUrl.getQueryParametersToExclude', function(&$parametersToExclude) {
+            if (!empty($this->referrerHost) && strpos($this->referrerHost, 'instagram.com') !== false) {
+                $parametersToExclude[] = 'e';
+                $parametersToExclude[] = 's';
+            }
+            if (!empty($this->referrerHost) && strpos($this->referrerHost, 'facebook.com') !== false) {
+                $parametersToExclude[] = 'h';
```

Comment:
@sgiehl can we also remove `fbclid` here or would this cause any issues with a possible future paid advertising integration for DB?

Reply:
`fbclid` is included in `url_query_parameter_to_exclude_from_url`

Reply:
👍  I see... noticed it in the log_visit table it is eg like `https://l.facebook.com/l.php?u=https://foo.bar/baz?fbclid=IwAR2RmCIQ_Nmn0K9V7B-c8BBMR8nBuEMVud5XW_tRMNP5rLZOcm2b9BsT1QE` so the fcblid seems to be actually part of the url in `u`... guess that's not really removable so easily.